pilot/pkg/model/authentication.go

```diff
@@ -182,37 +182,44 @@ func CollectJwtSpecs(policy *authn.Policy) []*authn.Jwt {
 // (https://github.com/envoyproxy/data-plane-api/pull/530/files).
 // The output of this function should use the Envoy data-plane-api proto once
 // this migration finished.
-func ConvertPolicyToJwtConfig(policy *authn.Policy) *mccpb.EndUserAuthenticationPolicySpec {
+func ConvertPolicyToJwtConfig(policy *authn.Policy) *jwtfilter.JwtAuthentication {
 	policyJwts := CollectJwtSpecs(policy)
 	if len(policyJwts) == 0 {
 		return nil
 	}
-	ret := &mccpb.EndUserAuthenticationPolicySpec{}
+	ret := &jwtfilter.JwtAuthentication{
+		AllowMissingOrFailed: true,
+	}
 	for _, policyJwt := range policyJwts {
 		hostname, port, _, err := ParseJwksURI(policyJwt.JwksUri)
 		if err != nil {
 			log.Errorf("Cannot parse jwks_uri %q: %v", policyJwt.JwksUri, err)
 			continue
 		}
-		jwt := &mccpb.JWT{
-			Issuer:                 policyJwt.Issuer,
-			Audiences:              policyJwt.Audiences,
-			JwksUri:                policyJwt.JwksUri,
-			ForwardJwt:             true,
-			PublicKeyCacheDuration: &types.Duration{Seconds: jwtPublicKeyCacheSeconds},
-			JwksUriEnvoyCluster:    JwksURIClusterName(hostname, port),
+
+		jwt := &jwtfilter.JwtRule{
+			Issuer:    policyJwt.Issuer,
+			Audiences: policyJwt.Audiences,
+			JwksSourceSpecifier: &jwtfilter.JwtRule_RemoteJwks{
+				RemoteJwks: &jwtfilter.RemoteJwks{
+					HttpUri: &core.HttpUri{
+						Uri: policyJwt.JwksUri,
+						HttpUpstreamType: &core.HttpUri_Cluster{
+							Cluster: JwksURIClusterName(hostname, port),
+						},
+					},
+					CacheDuration: &types.Duration{Seconds: jwtPublicKeyCacheSeconds},
+				},
+			},
+			Forward: true,
```

Comment:
can you move this code into networking/plugins/auth ? We try to keep envoy stuff away from model, so that we can refactor this piece easily.

Reply:
Can I do it in a next PR? This one is to follow https://github.com/istio/proxy/pull/1330 to switch to new proto so I try to minimize the changes so that we can submit them together and avoid keeping backward compatible.